conda/common/configuration.py

```diff
@@ -1314,6 +1314,21 @@ def __init__(cls, name, bases, attr):
             for name, p in cls.__dict__.items()
             if isinstance(p, ParameterLoader)
         )
+        cls.parameter_names_and_aliases = tuple(
+            name for p in cls.parameter_loaders.values() for name in p._names
+        )
+
+    @property
+    def parameter_loaders(cls):
```

Comment:
Move this to the `Configuration` class itself so we don't have to do `self.__class__.parameter_loaders`?

Reply:
hm we leave it as a metaclass property but alias it in the instance object?

```python
class ConfigurationType(type):
    @property
    def _parameter_loaders(cls) -> dict[str, ParameterLoader]:
        return {
            name: param
            for name, param in cls.__dict__.items()
            if isinstance(param, ParameterLoader)
        }

    def __call__(cls, *args, **kwargs):
        self = super().__call__(*args, **kwargs)
        self._parameter_loaders = cls._parameter_loaders
        return self
```

on reflection I think this is getting too far into the weeds and isn't helping with correcting the actual parameter aliasing issue, probably best to revert it to what you originally had?